alacritty_terminal/src/config/mod.rs

```diff
@@ -418,17 +398,35 @@ impl Default for DefaultTrueBool {
     }
 }
 
+fn defaulted<T, E>(err: E) -> T
+    where T: Default, E: Display
+{
+    error!("Problem with config: {}; using default value", err);
+    T::default()
+}
+
 pub fn failure_default<'a, D, T>(deserializer: D) -> Result<T, D::Error>
 where
     D: Deserializer<'a>,
     T: Deserialize<'a> + Default,
 {
-    let value = serde_yaml::Value::deserialize(deserializer)?;
-    match T::deserialize(value) {
-        Ok(value) => Ok(value),
-        Err(err) => {
-            error!("Problem with config: {}; using default value", err);
-            Ok(T::default())
-        },
-    }
+    Ok(T::deserialize(Value::deserialize(deserializer)?).unwrap_or_else(defaulted))
+}
+
+pub fn or_none<'de, T, D>(deserializer: D) -> Result<Option<T>, D::Error>
+    where D: Deserializer<'de>, T: Deserialize<'de> + Default
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(ref value) if value.to_lowercase() == "none" => None,
+        value => Some(T::deserialize(value).unwrap_or_else(defaulted))
+    })
+}
+
+pub fn struct_or_str<'de, T, D>(deserializer: D) -> Result<T, D::Error>
```

Comment:
If we want to make this generic, we should use it for the shell too. Otherwise it should be renamed and moved closer to where it's actually used.

Reply:
It won't work for the shell because of the `Option`, which will require a different function (without the `Default` bound, and defaulting to `None` on type error)

Reply:
It might be possible to get rid of the option for shell and use default instead, but if we don't want to use it we should change the name at least.

Reply:
I added a second function to handle the option case.  They'll probably be generically useful for future settings where you want to have an abbreviated string form or a more detailed full form.